packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -313,10 +313,17 @@ class _NestedScrollViewState extends State<NestedScrollView> {
     super.dispose();
   }
 
+  bool _lastHasScrolledBody;
+
   void _handleHasScrolledBodyChanged() {
     if (!mounted)
       return;
-    setState(() { /* _coordinator.hasScrolledBody changed (we use it in the build method) */ });
+    final bool newHasScrolledBody = _coordinator.hasScrolledBody;
+    if (_lastHasScrolledBody != newHasScrolledBody) {
+      setState(() {
+        // _coordinator.hasScrolledBody changed (we use it in the build method)
```

Comment:
Would it make sense to set `_lastHasScrolledBody` here (and call it `_hasScrolledBody`) instead of in the `build` method? Usually we try to mutate private state in `setState` instead of in build methods. If there's a good reason not to, maybe we can document it in a comment.

Reply:
I'll add a comment. The problem with setting it only in the setState is that the build method gets called more often than just when we call setState.